test/governance/extensions/GovernorSequentialProposalId.test.js

```diff
@@ -0,0 +1,202 @@
+
+          expect(this.mock.hashProposal(...this.proposal.shortProposal)).to.eventually.equal(this.proposal.hash);
+          await expect(this.mock.getProposalId(...this.proposal.shortProposal)).revertedWithCustomError(
+            this.mock,
+            'GovernorNonexistentProposal',
+          );
+          expect(this.mock.latestProposalId()).to.eventually.equal(i - 1);
+
+          await expect(this.helper.connect(this.proposer).propose())
+            .to.emit(this.mock, 'ProposalCreated')
+            .withArgs(
+              i,
+              this.proposer,
+              this.proposal.targets,
+              this.proposal.values,
+              this.proposal.signatures,
+              this.proposal.data,
+              anyValue,
+              anyValue,
+              this.proposal.description,
+            );
+
+          expect(this.mock.hashProposal(...this.proposal.shortProposal)).to.eventually.equal(this.proposal.hash);
+          expect(this.mock.getProposalId(...this.proposal.shortProposal)).to.eventually.equal(i);
+          expect(this.mock.latestProposalId()).to.eventually.equal(i);
+        }
+      });
+
+      it('sequential proposal ids with offset start', async function () {
+        const offset = 69420;
+        await this.mock.$_initializeLatestProposalId(offset);
+
+        for (const i of iterate.range(offset + 1, offset + 10)) {
+          this.proposal.description = `<proposal description #${i}>`;
+
+          expect(this.mock.hashProposal(...this.proposal.shortProposal)).to.eventually.equal(this.proposal.hash);
+          await expect(this.mock.getProposalId(...this.proposal.shortProposal)).revertedWithCustomError(
+            this.mock,
+            'GovernorNonexistentProposal',
+          );
+          expect(this.mock.latestProposalId()).to.eventually.equal(i - 1);
+
+          await expect(this.helper.connect(this.proposer).propose())
+            .to.emit(this.mock, 'ProposalCreated')
+            .withArgs(
+              i,
+              this.proposer,
+              this.proposal.targets,
+              this.proposal.values,
+              this.proposal.signatures,
+              this.proposal.data,
+              anyValue,
+              anyValue,
+              this.proposal.description,
+            );
+
+          expect(this.mock.hashProposal(...this.proposal.shortProposal)).to.eventually.equal(this.proposal.hash);
+          expect(this.mock.getProposalId(...this.proposal.shortProposal)).to.eventually.equal(i);
+          expect(this.mock.latestProposalId()).to.eventually.equal(i);
+        }
+      });
+
+      it('can only initialize proposal count from 0', async function () {
+        await this.helper.propose();
+        expect(this.mock.latestProposalId()).to.eventually.equal(1);
+        await expect(this.mock.$_initializeLatestProposalId(2)).to.be.revertedWithCustomError(
+          this.mock,
+          'GovernorAlreadyInitializedLatestProposalId',
+        );
+      });
```

Comment:
We need to update this test. Also, we're missing a test for the `GovernorInvalidLatestProposalId` error

Reply:
Yes good point. I actually prefer to remove the check for `if (newLatestProposalId == 0)` since its not fully necessary anymore. It was added when there was the nextProposalId concept and reducing it would cause issues. Here, the do nothing scenario is setting to 0 which isn't necessarily an issue.